### PR TITLE
Add support for s3credentials 401 redirects

### PR DESCRIPTION
**Pull request by bsatoriu**

### CODE REVIEW

**Comment by bsatoriu** [comment]
@chuckwondo my updates are ready for review.

File: api/cas/cas_auth.py [hunk 1]
```diff
@@ -249,8 +250,30 @@ def wrap(*args, **kwargs):
...
+        abort(status.HTTP_403_FORBIDDEN, description="Not authorized.")
 
     return wrap
 
 
+def edl_federated_request(url, stream_response=False):
+    s = requests.Session()
+    response = s.get(url, stream=stream_response)
+
+    if response.status_code == status.HTTP_401_UNAUTHORIZED:
```
Comment (chuckwondo) [inline api/cas/cas_auth.py:262]: If this is `False`, this function returns `None` instead of the `response` from the line preceding the `if`.

**Review by chuckwondo — CHANGES_REQUESTED** [review]

File: api/cas/cas_auth.py (hunk 1 above)
Comment (bsatoriu) [reply api/cas/cas_auth.py:262]: I removed the extraneous if conditions from this method

**Review by chuckwondo — APPROVED** [review]
Approving because a user is currently blocked without this fix, but I suggest adding unit tests for this, especially if you find that this either doesn't work after deploying, or that it breaks existing functionality.

**Pull request merged**